internal/api/oauthserver/service.go

```diff
@@ -168,13 +195,16 @@ func (s *Server) registerOAuthServerClient(ctx context.Context, params *OAuthSer
 	client.SetRedirectURIs(params.RedirectURIs)
 	client.SetGrantTypes(grantTypes)
 
-	// Generate client secret for all clients
-	plaintextSecret := generateClientSecret()
-	hash, err := hashClientSecret(plaintextSecret)
-	if err != nil {
-		return nil, "", errors.Wrap(err, "failed to hash client secret")
+	var plaintextSecret string
+	// Only generate client secret for confidential clients
+	if client.IsConfidential() {
+		plaintextSecret = generateClientSecret()
+		hash, err := hashClientSecret(plaintextSecret)
```

Comment:
`bcrypt` is only necessary if the secret is human-generated. Not necessary if the secret comes from a CSPRNG source... you can just use SHA256 then.